### PR TITLE
build(deps): upgrade flurl

### PR DESCRIPTION
**Pull request by Phil91**
## Description

* upgrade Flurl packages to the latest major
* remove Newtonsoft.Json references from all non test projects

## Why

To upgrade the project dependencies and remove newtonsoft json from non test projects

## Issue

Refs: #1052

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [x] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/keycloak/Keycloak.ErrorHandling/FlurlUntrustedCertExceptionHandler.cs [hunk 1]
```diff
@@ -30,25 +28,9 @@
     {
         foreach (var urlToTrust in urlsToTrust)
         {
-            FlurlHttp.ConfigureClient(urlToTrust, cli => cli.Settings.HttpClientFactory = new UntrustedCertHttpClientFactory());
+            FlurlHttp
+                .ConfigureClientForUrl(urlToTrust)
+                .ConfigureInnerHandler(chc => chc.ServerCertificateCustomValidationCallback = (_, _, _, _) => true);
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/keycloak/Keycloak.ErrorHandling/FlurlUntrustedCertExceptionHandler.cs:33]: This is only enabled on development environments

**Comment by Phil91** [comment]
IP Issues are created

[nuget/nuget/-/Flurl.Http.Signed/4.0.2](https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/16447)
[nuget/nuget/-/Flurl.Signed/4.0.0](https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/16448)

File: src/keycloak/Keycloak.Factory/KeycloakFactory.cs [hunk 2]
```diff
@@ -17,36 +17,25 @@
...
 {
-    private readonly KeycloakSettingsMap _settings;
+    private readonly KeycloakSettingsMap _settings = options.Value;
 
-    private static readonly JsonSerializerSettings SerializerSettings = new()
+    private static readonly JsonSerializerOptions SerializerSettings = new()
     {
-        NullValueHandling = NullValueHandling.Ignore,
-        ContractResolver = new DefaultContractResolver
-        {
```
Comment (ntruchsess) [inline src/keycloak/Keycloak.Factory/KeycloakFactory.cs:37]: did you validate Dictionaries are deserialized preserving the keys? (preserving the case of dictionary-keys was added in https://github.com/eclipse-tractusx/portal-backend/pull/910 and is required to correctly deserialize configuration-data that is not defined as model-class but deserializes into a dictionary. Not preserving the case does break keycloak-seeding).

**Review by ntruchsess — CHANGES_REQUESTED** [review]
see comments

File: src/keycloak/Keycloak.Factory/KeycloakFactory.cs (hunk 2 above)
Comment (Phil91) [reply src/keycloak/Keycloak.Factory/KeycloakFactory.cs:37]: I've added the test GetIdentityProviderMappers_WithFilledDictionary_ReturnsExpected to check if the keys are preserved which is the case

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**